encoding/json/inbound_test.go

```diff
@@ -76,6 +76,27 @@ func TestHandleMapSuccess(t *testing.T) {
 	assert.Equal(t, "true", response.Success)
 }
 
+func TestHandleSuccessWithResponseHeaders(t *testing.T) {
+	h := func(*Request, *simpleRequest) (*simpleResponse, *Response, error) {
+		response := &Response{Headers: transport.Headers{"foo": "bar"}}
+		return &simpleResponse{Success: true}, response, nil
+	}
+
+	handler := jsonHandler{
+		reader:  structReader{reflect.TypeOf(simpleRequest{})},
+		handler: reflect.ValueOf(h),
+	}
+
+	resw := new(transporttest.FakeResponseWriter)
+	err := handler.Handle(context.Background(), &transport.Request{
+		Procedure: "simpleCall",
+		Body:      jsonBody(`{"name": "foo", "attributes": {"bar": 42}}`),
+	}, resw)
+	require.NoError(t, err)
+
+	assert.Equal(t, transport.Headers{"foo": "bar"}, resw.Headers)
```

Comment:
optional: should we retype `transport.Headers` so that a user doesn't need to interact with that package in any way.
and from an efficiency perspective, we would just cast it since they're the same.